src/legacy/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -226,7 +226,10 @@ export const buildPipelineVisFunction: BuildPipelineVisFunction = {
   },
   markdown: visState => {
     const { markdown, fontSize, openLinksInNewTab } = visState.params;
-    const escapedMarkdown = escapeString(markdown);
+    let escapedMarkdown = '';
+    if (typeof markdown === 'string' || markdown instanceof String) {
```

Comment:
Nothing prevents future changes to the markdown code to save null, undefined or numbers to the current markdown string object, so one check more is better than one more error.
There can also be the rare case where you changed manually the URL of a markdown and you created a malformed url like this:
```
http://localhost:5601/xhr/app/kibana#/visualize/create?type=markdown&_g=()&_a=(filters:!(),linked:!f,query:(language:kuery,query:''),uiState:(),vis:(aggs:!(),params:(fontSize:12,markdown:123,openLinksInNewTab:!f),title:'',type:markdown))
```
where the markdown test is a number instead of a string. Without that check, it will crash the page. In this case the markdown is also not visualized. 
Similar if the url is changed with a boolean value.
As written: it's a rare case, but it's at least one more safety check for the escapeString function until we don't have a fully TS implementation of our code.

If you think it's not necessary I can remove that.

Reply:
Makes sense! I don't think we need to remove it, mostly I was just curious about the cases you had in mind